test/parallel/test-assert-objects.js

```diff
@@ -0,0 +1,384 @@
+    const obj2 = createCircularObject();
+    assert.matchObjectStrict(obj1, obj2);
+  });
+
+  it('should not strictly compare two objects with different circular references', function() {
+    const obj1 = createCircularObject();
+    const obj2 = { self: {} };
+    assert.throws(() => assert.matchObjectStrict(obj1, obj2), Error);
+  });
+
+  it('should loosely compare two objects with circular references', function() {
+    const obj1 = createCircularObject();
+    const obj2 = createCircularObject();
+    assert.matchObject(obj1, obj2);
+  });
+
+  it('should strictly compare two arrays with identical elements', function() {
+    const arr1 = [1, 'two', true];
+    const arr2 = [1, 'two', true];
+    assert.matchObjectStrict(arr1, arr2);
+  });
+
+  it('should not strictly compare two arrays with different elements', function() {
+    const arr1 = [1, 'two', true];
+    const arr2 = [1, 'two', false];
+    assert.throws(() => assert.matchObjectStrict(arr1, arr2), Error);
+  });
+
+  it('should loosely compare two arrays with type coercion', function() {
+    const arr1 = [1, '2', true];
+    const arr2 = [1, 2, 1];
+    assert.matchObject(arr1, arr2);
+  });
+
+  it('should strictly compare two Date objects with the same time', function() {
+    const date1 = new Date('2024-06-10T12:00:00Z');
+    const date2 = new Date('2024-06-10T12:00:00Z');
+    assert.matchObjectStrict(date1, date2);
+  });
+
+  it('should not strictly compare two Date objects with different times', function() {
+    const date1 = new Date('2024-06-10T12:00:00Z');
+    const date2 = new Date('2024-06-11T12:00:00Z');
+    assert.throws(() => assert.matchObjectStrict(date1, date2), Error);
+  });
+
+  it('should strictly compare two objects with large number of properties', function() {
+    const obj1 = Array.from({ length: 100 }, (_, i) => ({
+      [`key${i}`]: i,
+    })).reduce((acc, cur) => ({ ...acc, ...cur }), {});
+    const obj2 = Array.from({ length: 100 }, (_, i) => ({
+      [`key${i}`]: i,
+    })).reduce((acc, cur) => ({ ...acc, ...cur }), {});
+    assert.matchObjectStrict(obj1, obj2);
+  });
+
+  it('should not strictly compare two objects with different large number of properties', function() {
+    const obj1 = Array.from({ length: 100 }, (_, i) => ({
+      [`key${i}`]: i,
+    })).reduce((acc, cur) => ({ ...acc, ...cur }), {});
+    const obj2 = Array.from({ length: 100 }, (_, i) => ({
+      [`key${i}`]: i + 1,
+    })).reduce((acc, cur) => ({ ...acc, ...cur }), {});
+    assert.throws(() => assert.matchObjectStrict(obj1, obj2), Error);
+  });
+
+  it('should strictly compare two objects with Symbol properties', function() {
+    const sym = Symbol('test');
+    const obj1 = { [sym]: 'symbol' };
+    const obj2 = { [sym]: 'symbol' };
+    assert.matchObjectStrict(obj1, obj2);
+  });
+
+  it('should not strictly compare two objects with different Symbols', function() {
+    const sym1 = Symbol('test1');
+    const sym2 = Symbol('test2');
```

Comment:
Worth validating that using two symbols with the same description does not change the test
```suggestion
    const sym1 = Symbol('test');
    const sym2 = Symbol('test');
```